binsync/decompilers/ghidra/server/controller.py

```diff
@@ -126,5 +126,15 @@ def functions(self, **kwargs) -> dict:
             return None
         funcs = {}
         for addr in ret:
-            funcs[addr] = Function(addr, ret[addr]["size"], header=FunctionHeader(ret[addr]["name"], addr))        
+            stack_vars = {}
+            for offset in ret[addr]["stack_vars"]:
+                sv = StackVariable(None, None, None, None, None)
+                stack_vars[offset] = sv.__setstate__(ret[addr]["stack_vars"][offset])
+            funcs[addr] = Function(addr, ret[addr]["size"], header=FunctionHeader(ret[addr]["name"], addr), stack_vars=stack_vars)
         return funcs
+
+    def stack_vars(self, addr, **kwargs) -> dict:
+        ret = self.ghidra.get_stack_vars()
```

Comment:
You should also always return an Artifact Type, which will require you to convert the dictionary like you did above. For now I recommend a helper function that does that conversion so you dont duplicate code.

Reply:
Sounds good

Reply:
Is there a particular place you put helpers, or is at the bottom with the other functions fine

Reply:
@Flipout50 yeah you can just put it on the bottom